ch_tools/monrun_checks/ch_backup.py

```diff
@@ -22,41 +21,79 @@
     default=3,
-    help="Critical threshold for failed backups.",
+    help="Critical threshold on the number of failed backups. If last backups failed and its count equals or greater "
+    "than the specified threshold, a crit will be reported.",
 )
 @click.option(
-    "--critical-absent",
-    "crit_absent",
+    "--backup-age-warn",
+    "--backup-age-warn-threshold",
+    "backup_age_warn_threshold",
+    default="1d",
+    type=TimeSpanParamType(),
+    help="Warning threshold on age of the last backup. If the last backup is created more than the specified "
+    "time ago, a warning will be reported.",
+)
+@click.option(
+    "--backup-age-crit",
+    "--backup-age-crit-threshold",
+    "--critical-absent",  # deprecated option name preserved for backward-compatibility
+    "backup_age_crit_threshold",
     default="2d",
     type=TimeSpanParamType(),
-    help="Critical threshold for absent backups.",
+    help="Critical threshold on age of the last backup. If the last backup is created more than the specified "
+    "time ago, a crit will be reported.",
 )
-def backup_command(crit_failed, crit_absent):
+@click.option(
+    "--backup-count-warn",
+    "--backup-count-warn-threshold",
+    "backup_count_warn_threshold",
+    default=0,
+    help="Warning threshold on the number of backups. If the number of backups of any status equals or greater than "
+    "the specified threshold, a warning will be reported.",
+)
+@click.option(
+    "--min-uptime",
+    "min_uptime",
+    default="2d",
+    type=TimeSpanParamType(),
+    help="Minimal ClickHouse uptime enough for backup creation.",
+)
+def backup_command(
+    failed_backup_count_crit_threshold,
+    backup_age_warn_threshold,
+    backup_age_crit_threshold,
+    backup_count_warn_threshold,
+    min_uptime,
+):
     """
     Check ClickHouse backups: its state, age and count.
-    Skip backup checks for recently created clusters.
     """
-    dbaas_config = DbaasConfig.load()
-
-    cluster_created_at = parse_str_datetime(dbaas_config.created_at)
-    if not check_now_pass_threshold(cluster_created_at):
-        return
-
     ch_client = ClickhouseClient()
-    backup_config = BackupConfig.load()
     backups = get_backups()
 
-    check_restored_parts()
+    check_restored_data()
+    check_uptime(ch_client, min_uptime)
     check_valid_backups_exist(backups)
-    check_last_backup_not_failed(backups, crit=crit_failed)
-    check_backup_age(ch_client, backups, crit=crit_absent)
-    check_backup_count(backup_config, backups)
+    check_last_backup_not_failed(backups, failed_backup_count_crit_threshold)
+    check_backup_age(backups, backup_age_warn_threshold, backup_age_crit_threshold)
+    check_backup_count(backups, backup_count_warn_threshold)
+
+
+def check_uptime(ch_client: ClickhouseClient, min_uptime: timedelta) -> None:
+    """
+    Check that ClickHouse uptime enough for backup creation.
+    """
+    uptime = ch_client.get_uptime()
```

Comment:
maybe we need to try-catch clickhouse connection errors there, if Clickhouse is dead?
Or it will be suppressed somewhere else?

Reply:
It will be handled in https://github.com/yandex/ch-tools/blob/25728d2e3287c2e19d9c14259a6f466ddfe38c3a/ch_tools/monrun_checks/main.py#L83 and resulted to a warning. 

But I reworked the implementation to handle this scenario more explicitly and report more clear message.